plugins/techdocs/src/reader/components/TechDocsReaderPage/TechDocsReaderPage.test.tsx

```diff
@@ -76,6 +82,24 @@ const techdocsStorageApiMock: jest.Mocked<typeof techdocsStorageApiRef.T> = {
   syncEntityDocs: jest.fn(),
 };
 
+const storageApiMock = MockStorageApi.create();
+
+const discoveryApiMock = {
+  getBaseUrl: jest
+    .fn()
+    .mockResolvedValue('https://localhost:7000/api/techdocs'),
+};
+
+const fetchApiMock = {
+  fetch: jest.fn().mockResolvedValue({
```

Comment:
This works fine, but if you like in the future, we have a preference toward using `msw` ([example](https://github.com/backstage/backstage/blob/fb9efa95a19451f2cf8f753eda400a921b2edaa2/packages/backend-common/src/reading/BitbucketUrlReader.test.ts#L107) and [usage](https://github.com/backstage/backstage/blob/fb9efa95a19451f2cf8f753eda400a921b2edaa2/packages/backend-common/src/reading/BitbucketUrlReader.test.ts#L308)) a lot instead of mocking `fetch`. That has the benefit of strictly locking down the test runner, so that no network calls can accidentally leave it, and no need to deal with the sort of sometimes complex-to-mock `fetch` interface. Also, it'll by design automatically ensure that the actual URL used when doing the request is the right one.

Same goes for the other tests

Reply:
That is a great idea! Do you mind if I do that in a follow-up pull-request?

Reply:
Absolutely ok